adafruit_bno055.py

```diff
@@ -327,6 +389,184 @@ def gravity(self):
+        value = self._read_register(_GYRO_CONFIG_0_REGISTER)
+        masked_value = 0b00000111 & value
+        self._write_register(_GYRO_CONFIG_0_REGISTER, masked_value | bandwidth)
+        self._write_register(_PAGE_REGISTER, 0x00)
+
+    @property
+    def gyro_mode(self):
+        """ Switch the gyroscope mode and return the new mode. Default value: Normal
+        See table 3-9 in the datasheet.
+        """
+        self._write_register(_PAGE_REGISTER, 0x01)
+        value = self._read_register(_GYRO_CONFIG_1_REGISTER)
+        self._write_register(_PAGE_REGISTER, 0x00)
+        return 0b00000111 & value
+
+    @gyro_mode.setter
+    def gyro_mode(self, mode=GYRO_NORMAL_MODE):
+        if self.mode in [0x08, 0x09, 0x0A, 0x0B, 0x0C]:
+            raise RuntimeError("Mode must not be a fusion mode")
+        self._write_register(_PAGE_REGISTER, 0x01)
+        value = self._read_register(_GYRO_CONFIG_1_REGISTER)
+        masked_value = 0b00000000 & value
+        self._write_register(_GYRO_CONFIG_1_REGISTER, masked_value | mode)
+        self._write_register(_PAGE_REGISTER, 0x00)
+
+    @property
+    def magnet_rate(self):
+        """ Switch the magnetometer data output rate and return the new rate. Default value: 20Hz
+        See table 3-10 in the datasheet.
+        """
+        self._write_register(_PAGE_REGISTER, 0x01)
+        value = self._read_register(_MAGNET_CONFIG_REGISTER)
+        self._write_register(_PAGE_REGISTER, 0x00)
+        return 0b00000111 & value
+
+    @magnet_rate.setter
+    def magnet_rate(self, rate=MAGNET_20HZ):
+        if self.mode in [0x08, 0x09, 0x0A, 0x0B, 0x0C]:
+            raise RuntimeError("Mode must not be a fusion mode")
+        self._write_register(_PAGE_REGISTER, 0x01)
+        value = self._read_register(_MAGNET_CONFIG_REGISTER)
+        masked_value = 0b01111000 & value
+        self._write_register(_MAGNET_CONFIG_REGISTER, masked_value | rate)
+        self._write_register(_PAGE_REGISTER, 0x00)
+
+    @property
+    def magnet_operation_mode(self):
+        """ Switch the magnetometer operation mode and return the new mode. Default value: Regular
+        See table 3-10 in the datasheet.
+        """
+        self._write_register(_PAGE_REGISTER, 0x01)
+        value = self._read_register(_MAGNET_CONFIG_REGISTER)
+        self._write_register(_PAGE_REGISTER, 0x00)
+        return 0b00011000 & value
+
+    @magnet_operation_mode.setter
+    def magnet_operation_mode(self, mode=MAGNET_REGULAR_MODE):
+        if self.mode in [0x08, 0x09, 0x0A, 0x0B, 0x0C]:
+            raise RuntimeError("Mode must not be a fusion mode")
+        self._write_register(_PAGE_REGISTER, 0x01)
+        value = self._read_register(_MAGNET_CONFIG_REGISTER)
+        masked_value = 0b01100111 & value
+        self._write_register(_MAGNET_CONFIG_REGISTER, masked_value | mode)
+        self._write_register(_PAGE_REGISTER, 0x00)
+
+    @property
+    def magnet_power_mode(self):
```

Comment:
How about naming this one just `magnet_mode` to better match `accel_mode` and `gyro_mode`? It looks like all three of those are functionally the same thing for each sensor.

And then leave `magnet_operation_mode` as is. That one looks unique to the magnet.

Reply:
Yeah, that makes sense